sockeye/inference.py

```diff
@@ -1441,6 +1473,12 @@ def _beam_search(self,
             # finished rows are inf everywhere except column zero, which holds the accumulated model score
             scores = self._update_scores.forward(scores, finished, inactive, scores_accumulated, self.inf_array, pad_dist)
 
+            # Mark entries that should be blocked as having a score of np.inf
+            if self.global_avoid_trie or any(raw_avoid_list):
+                indices = mx.nd.array(avoid_states.avoid(), ctx=self.context)
+                if any(indices):
+                    scores.reshape((self.batch_size * self.beam_size * scores.shape[-1]))[indices] = np.inf
```

Comment:
This looks strange. Is `scores` after this line reshaped or not?

Reply:
It isn't reshaped. Here is the problem: say word ID 74 is always blocked, batch_size is 1, and beam_size is 5, and the maximum vocabulary size is 10,000. We need to block ID 74 in every row of `scores` (by setting it to `np.inf`). The easiest way to do this is to reshape `scores` into a single-dimension array, and block ID 74, ID 10074, 20074, 30074, and 40074. That's what this code does.

Reply:
I understand what the code does, but was just wondering about the clarity of this expression (using fluent reshape and inplace setting of values).

Reply:
I see. 2d indices in the first place would make more sense, if MXNet supports that.